elastic/kibana
# [Stateful sidenav] Put new nav behind launch darkly flag

In this PR I've replaced the `kibana.yml` feature flag with a Launch Darkly feature flag. The Launch Darkly flag will only be enabled if both these conditions are met:

- Kibana is running in QA environment
- The deployment belongs to an Elastic user

The following `kibana.yml` configs were removed

```yaml
navigation.solutionNavigation.featureOn
navigation.solutionNavigation.optInStatus
```

I've also cleaned up the Kibana Global settings and only left the toggle to enable the new navigation. This toggle is only visible in the UI if the Launch Darkly flag is enabled.

**Important:** This PR does **not** yet interact with LD. It is the preparatory work to read the feature flag.

## How to test

* Add the following to your `kibana.dev.yml` file

```yaml
xpack.cloud_integrations.experiments.enabled: true
xpack.cloud_integrations.experiments.flag_overrides:
  "navigation.solutionNavEnabled": true

xpack.cloud.id: "ftr_fake_cloud_id:aGVsbG8uY29tOjQ0MyRFUzEyM2FiYyRrYm4xMjNhYmM="
xpack.cloud.base_url: "https://cloud.elastic.co"
xpack.cloud.deployment_url: "/deployments/deploymentId"
```

* Start Kibana
* The navigation should still be the old one
* Navigate to Management > Kibana > Settings
* Under the Global settings section, you should see a toggle to enable the new navigation

## Global setting available when the feature is on

<img width="1098" alt="Screenshot 2024-04-04 at 10 15 23" src="https://github.com/elastic/kibana/assets/2854616/f3c706d2-40e7-4a28-b148-3c862180d7cf">
